gapic-generator-java/src/main/java/com/google/api/generator/gapic/composer/common/AbstractTransportServiceStubClassComposer.java

```diff
@@ -1233,12 +1251,200 @@ protected TypeNode getTransportOperationsStubType(Service service) {
     return transportOpeationsStubType;
   }
 
+  protected static LambdaExpr createRequestMutatorClassInstance(
+      Method method, ImmutableMap<String, Message> messageTypes) {
+    List<Statement> bodyStatements = new ArrayList<>();
+    VariableExpr requestVarExpr = createRequestVarExpr(method);
+
+    Reference requestBuilderRef =
+        VaporReference.builder()
+            .setEnclosingClassNames(method.inputType().reference().name())
+            .setName("Builder")
+            .setPakkage(method.inputType().reference().pakkage())
+            .build();
+
+    TypeNode requestBuilderType = TypeNode.withReference(requestBuilderRef);
+
+    VariableExpr requestBuilderVarExpr =
+        VariableExpr.builder()
+            .setVariable(
+                Variable.builder().setName("requestBuilder").setType(requestBuilderType).build())
+            .setIsDecl(true)
+            .build();
+
+    MethodInvocationExpr setRequestBuilderInvocationExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(requestVarExpr)
+            .setMethodName("toBuilder")
+            .setReturnType(requestBuilderType)
+            .build();
+
+    Expr requestBuilderExpr =
+        AssignmentExpr.builder()
+            .setVariableExpr(requestBuilderVarExpr)
+            .setValueExpr(setRequestBuilderInvocationExpr)
+            .build();
+
+    bodyStatements.add(ExprStatement.withExpr(requestBuilderExpr));
+
+    VariableExpr returnBuilderVarExpr =
+        VariableExpr.builder()
+            .setVariable(
+                Variable.builder().setName("requestBuilder").setType(requestBuilderType).build())
+            .setIsDecl(false)
+            .build();
+
+    MethodInvocationExpr.Builder returnExpr =
+        MethodInvocationExpr.builder()
+            .setExprReferenceExpr(returnBuilderVarExpr)
+            .setMethodName("build");
+
+    createRequestMutatorBody(method, messageTypes, bodyStatements, returnBuilderVarExpr);
+
+    return LambdaExpr.builder()
+        .setArguments(requestVarExpr.toBuilder().setIsDecl(true).build())
+        .setBody(bodyStatements)
+        .setReturnExpr(returnExpr.build())
+        .build();
+  }
+
+  @VisibleForTesting
+  static List<Statement> createRequestMutatorBody(
+      Method method,
+      ImmutableMap<String, Message> messageTypes,
+      List<Statement> bodyStatements,
+      VariableExpr returnBuilderVarExpr) {
+
+    if (!shouldGenerateRequestMutator(method, messageTypes)) {
+      return bodyStatements;
+    }
```

Comment:
qq, why is `createRequestMutatorBody()` gated with a `shouldGenerateRequestMutator()` check?

I believe there was a check above: `method.hasAutoPopulatedFields() && shouldGenerateRequestMutator(method, messageTypes)` that checks if the callsettings should add a request mutator.

Reply:
Good q! The `shouldGenerateRequestMutator` method contains all of the null checks that were originally within this `createRequestMutatorBody` method. While the current code does as you say - e.g. it checks `shouldGenerateRequestMutator()` first, I added in this part more as a safety check in case `createRequestMutatorBody` would ever be used outside of the `shouldGenerateRequestMutator()` check. 

There might be a better way to do this or it might not be necessary - let me know if you think I should remove. Thanks!

Reply:
My initial thoughts are that we probably don't need it mostly because I would think that `createRequestMutatorBody` should be clear enough that it's creating request mutator body and using it outside of the request mutator logic section wouldn't make any sense. 

I don't think adding this safety check hurts so I'll leave it up to you.

Reply:
Removed it per offline discussion